rero_ils/modules/documents/dumpers/indexer.py

```diff
@@ -0,0 +1,228 @@
+        # "nested" field using the `copy_to` directive ; this will cause an
+        # exception during index creation.
+        # Best solution seems to "script" this copy into the listener
+        data['nested_identifiers'] = data['identifiedBy']
+
+        # create specific keys for some common identifier families. It could
+        # be used as a shortcut to search specific identifiers for expert
+        # search mode.
+        identifier_families = {
+            'isbn': [IdentifierType.ISBN],
+            'issn': [IdentifierType.ISSN, IdentifierType.L_ISSN]
+        }
+        for key, family_types in identifier_families.items():
+            if filtered_identifiers := list(
+                {
+                    identifier.normalize()
+                    for identifier in identifiers
+                    if identifier.type in family_types
+                }
+            ):
+                data[key] = filtered_identifiers
+
+    def dump(self, record, data):
+        """Dump a document instance with basic document informations.
+
+        :param record: The record to dump.
+        :param data: The initial dump data passed in by ``record.dumps()``.
+        """
+        from rero_ils.modules.local_fields.api import LocalField
+        from rero_ils.utils import language_mapping
+
+        from ..api import Document
+        self.process_holdings(record, data)
+
+        if contributions := process_literal_contributions(
+            data.get('contribution', [])
+        ):
+            data.pop('contribution', None)
+            data['contribution'] = contributions
+
+        # TODO: compare record with those in DB to check which authors have
+        # to be deleted from index
+
+        # Index host document title in child document (part of)
+        for part_of in data.get('partOf', []):
+            doc_pid = part_of.get('document', {}).get('pid')
+            document = Document.get_record_by_pid(doc_pid).dumps()
+            if titles := [
+                v['_text']
+                for v in document.get('title', {})
+                if v.get('_text') and v.get('type') == 'bf:Title'
+            ]:
+                part_of['document']['title'] = titles.pop()
+
+        # sort title
+        sort_title = TitleExtension.format_text(
+            data.get('title', []),
+            with_subtitle=True
+        )
+        language = language_mapping(data.get('language')[0].get('value'))
+        if current_app.config.get('RERO_ILS_STOP_WORDS_ACTIVATE', False):
+            sort_title = current_app.\
+                            extensions['reroils-normalizer-stop-words'].\
+                            normalize(sort_title, language)
+        data['sort_title'] = sort_title
+
+        # Local fields in JSON
+        local_fields = LocalField.get_local_fields_by_resource(
+            'doc', record['pid'])
+        if local_fields:
+            data['local_fields'] = local_fields
+
+        self.process_identifiers(record, data)
```

Comment:
as `process_identifiers` is a static method you should use `IndexerDumper.process_identifiers(...)` even if `self` works (but using `self` breaks the concept of staticmethod)